examples/ucp_client_server.c

```diff
@@ -95,6 +96,95 @@ static struct {
  */
 static void usage(void);
 
+void buffer_free(ucp_dt_iov_t *iov)
+{
+    size_t idx;
+
+    for (idx = 0; idx < iov_cnt; idx++) {
+        if (iov[idx].buffer == NULL) {
+            continue;
+        }
+        mem_type_free(iov[idx].buffer);
+        iov[idx].buffer = NULL;
+        iov[idx].length = 0;
+    }
+    mem_type_free(iov);
+}
+
+ucp_dt_iov_t *buffer_malloc()
+{
+    size_t       idx;
+    ucp_dt_iov_t *iov;
+
+    iov = calloc(iov_cnt, sizeof(ucp_dt_iov_t));
+    CHKERR_ACTION(iov == NULL, "allocate memory\n", return NULL;);
+
+    for (idx = 0; idx < iov_cnt; idx++) {
+        iov[idx].length = test_string_length;
+        iov[idx].buffer = mem_type_malloc(iov[idx].length);
+        if (iov[idx].buffer == NULL) {
+            buffer_free(iov);
```

Comment:
need to release previous IOV buffers

Reply:
@yosefe ```buffer_free``` will release all the already allocated space.